### ISSUE TITLE
when I run it in doocker and use the chinese model ,it have this question :2021-04-16 05:52:17 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder 2021-04-16 05:52:18 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder

### ISSUE THREAD

**Issue by jichaoqun** [alumae/kaldi-gstreamer-server#253]

**Comment by jichaoqun** [comment]
t: \u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5
2021-04-16 05:51:58 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Postprocessing (final=False) result..
2021-04-16 05:51:58 -   DEBUG:       root: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for postprocessor lock
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Got partial result: \u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Got partial result: \u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Got final result: \u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Got full final result: {"total-length": 5.472, "segment-length": 5.45, "status": 0, "segment-start": 0.0, "result": {"final": true, "hypotheses": [{"likelihood": 1251.02, "transcript": "\u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5"}]}}
2021-04-16 05:51:58 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Before postprocessing: {u'status': 0, u'segment-start': 0.0, u'segment-length': 5.45, u'total-length': 5.472, u'result': {u'hypotheses': [{u'transcript': u'\u4eca\u5929 \u75c5\u4eba \u60c5\u51b5 \u826f\u597d \u6ca1\u6709 \u51fa\u73b0 \u53d1\u70ed \u7b49 \u60c5\u51b5', u'likelihood': 1251.02}], u'final': True}, 'segment': 0, 'id': u'7404beee-0d39-4d67-963c-01c58da10193'}
2021-04-16 05:51:58 -   DEBUG:       root: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for postprocessor lock
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Pipeline received eos signal
2021-04-16 05:51:58 -    INFO:   decoder2: 7404beee-0d39-4d67-963c-01c58da10193: Resetting decoder state
2021-04-16 05:51:58 -   DEBUG:       root: Waiting until processing threads finish (7)
2021-04-16 05:51:59 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:00 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:01 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:02 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:03 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:04 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:05 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:06 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:07 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:08 -   DEBUG:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Checking that decoder hasn't been silent for more than 10 seconds
2021-04-16 05:52:09 - WARNING:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: More than 10 seconds from last decoder hypothesis update, cancelling
2021-04-16 05:52:09 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Master disconnected before decoder reached EOS?
2021-04-16 05:52:09 -    INFO:   decoder2: <undefined>: Sending EOS to pipeline in order to cancel processing
2021-04-16 05:52:09 -    INFO:   decoder2: <undefined>: Cancelled pipeline
2021-04-16 05:52:09 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:10 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:11 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:12 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:13 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:14 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:15 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:16 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:17 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder
2021-04-16 05:52:18 -    INFO:   __main__: 7404beee-0d39-4d67-963c-01c58da10193: Waiting for EOS from decoder


I think this question that the work can decode the redio but it can not send the word to server ? maybe it others questions ?
in short Ican not solve it , who can tell me what i should do? thanks very much!!!

**Comment by jichaoqun** [comment]
should change the worker.py. in the 423 line:
    thread.start_new_thread(main_loop, (args.uri, decoder_pipeline, post_processor, full_post_processor))  
change to :
        thread.start_new_thread(main_loop, (args.uri, decoder_pipeline, None,None))
